src/package.rs

```diff
@@ -1,24 +1,45 @@
-#[derive(Serialize, Debug, Clone)]
+use std::fs::File;
+use std::io::Read;
+use std::error::Error;
+use std::sync::Mutex;
+
+use toml::Value;
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
 pub struct Package {
     name: String,
     category: String,
 }
 
 impl Package {
-    pub fn new(name: &str, category: &str) -> Package {
+    pub fn new(name: &str, category: &mut str) -> Package {
         Package {
             name: name.to_string(),
             category: category.to_string(),
         }
     }
 }
 
-// This is here for debugging purposes until we have a nice database
 lazy_static! {
-    pub static ref PACKAGE_LIST: Vec<Package> = vec![
-        Package::new("dash", "shell"),
-        Package::new("gcc-base", "sys-lib"),
-        Package::new("libc", "sys-lib"),
-        Package::new("libgcc", "sys-lib"),
-    ];
+    pub static ref PACKAGE_LIST: Mutex<Vec<Package>> = Mutex::new(vec![]);
+}
+
+pub fn load_packages() -> Result<(), Box<Error>> {
+    let mut file = File::open("packages.toml")?;
+
+    let mut packages = String::new();
+    file.read_to_string(&mut packages)?;
+
+    let toml = packages.parse::<Value>()?;
+
+    for (name, category) in toml.as_table().unwrap().iter() {
+        for (_, value) in category.as_table().unwrap().iter() {
```

Comment:
You are doing the parsing the wrong way.

An important idea to keep in mind when doing parsing is **to handle errors gracefully**, that should be your motto.

From the file you have commited, we can see the pattern you are parsing looks like this:

```
[package_name]
category = "category_name"
```

And what your parsing is doing is basically this:

for table in the toml file (these are the packages)
  for entries in table (these are all the lines within the package, like `category`)
     create a new package with the bigger table name and the entry value

This means that the given file:
```
[package1]
category = "baz"
foo = "bar"
```

Will in fact create two packages (and not one) : `baz/package1` and `package1/bar`.

What you should do is:
* Ensuring there is no more, no less informations in the file (No extra table like the `foo = bar` in the previous example)
* Ensuring all informations that should be named with a specific name are in fact named with their specific name.

Remember, **never trust user input**. Make a parser that is bullet proof, and when you're sure it is, then dumb-test it until you have proven yourself  it's not.